File: ModuleManeger/server/app/internal/module/schemas/modules.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any

class ContainerStatus(BaseModel):
	name: str
	state: str
	status: str

class ExemplStatus(BaseModel):
	containers: List[ContainerStatus]
	all_running: bool

# class ContaiderConfig(BaseModel):
# 	build: Optional[Dict[str, Any]] = None
# 	command: Optional[str] = None
# 	restart: str = ""
# 	environment: Dict[str, Any] = {}
# 	labels: List[str] = []
# 	volumes: List[str] = []

class ContaiderData(BaseModel):
	name: str
	config: Dict[str, Any]

class ModulesConf(BaseModel):
	name: str
	name_module: str
	multiply: bool
	containers: Optional[List[ContaiderData]] = None
	type: str
	dependencies: Optional[List[str]] = None

class ModulesLoadData(BaseModel):
	name: str
	path: Optional[str]
	status: Optional[ExemplStatus] = None

class ModulesConfAndLoad(ModulesConf):
	repo: Optional[str] = None
	local: bool = False
	load: bool = False
	load_module_name: List[ModulesLoadData] = []

class ModuleData(BaseModel):
	module: str
	exemle: str
	path: str
	config: ModulesConf

class AllModulesResData(BaseModel):
	data: List[ModulesConfAndLoad]

class ContainerId(BaseModel):
	container_id: str
	name: str
	service: str

class Containers(BaseModel):
	data: List[ContainerId]

class RebuildData(BaseModel):
	service: str
	name: str

File: frontend/client_base/src/entites/moduleManager/modules/modules.ts

export interface ContainerConfig{
    build?: Record<string, unknown>
	command: string
	restart: string
	environment: Record<string, unknown>
	labels: string[]
	volumes: string[]
}

export interface Containers{
    name: string
    confog: ContainerConfig
}

export interface Exemple{
    name: string
    path: string
    status?: {
        containers: {
            name: string
            state: string
            status: string
        }[],
        all_running: boolean
    }
}

export interface ModuleData{
    name: string
    name_module: string
    multiply: boolean
    containers: Containers[]
    load: boolean
    repo?: string
    local: boolean
    load_module_name: Exemple[]
}

export type AllModulesData = {
    data: ModuleData[]
}

export interface CoreContainerId{
    container_id: string
    name: string
    service: string
}

export type CoreContainerData = {
    data: CoreContainerId[]
}